apps/angular/1-projection/src/app/ui/card/card.component.ts

```diff
@@ -12,47 +8,38 @@ import { ListItemComponent } from '../list-item/list-item.component';
     <div
       class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
       [class]="customClass()">
-      @if (type() === CardType.TEACHER) {
-        <img ngSrc="assets/img/teacher.png" width="200" height="200" />
-      }
-      @if (type() === CardType.STUDENT) {
-        <img ngSrc="assets/img/student.webp" width="200" height="200" />
-      }
-
+      <img [ngSrc]="cardImgSrc()" width="200" height="200" priority />
       <section>
         @for (item of list(); track item) {
-          <app-list-item
-            [name]="item.firstName"
-            [id]="item.id"
-            [type]="type()"></app-list-item>
+          <app-list-item [id]="item.id" (delete)="delete.emit($event)">
+            <ng-container
+              *ngTemplateOutlet="
+                nameTemplate() || defaultName;
+                context: { $implicit: item }
+              "></ng-container>
+          </app-list-item>
         }
       </section>
 
       <button
         class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
+        (click)="addNewItem.emit()">
         Add
       </button>
+
+      <ng-template #defaultName let-item>
+        <span class="name">{{ item.firstName }}</span>
+      </ng-template>
     </div>
   `,
-  imports: [ListItemComponent, NgOptimizedImage],
+  imports: [ListItemComponent, NgOptimizedImage, NgTemplateOutlet],
 })
 export class CardComponent {
-  private teacherStore = inject(TeacherStore);
-  private studentStore = inject(StudentStore);
-
   readonly list = input<any[] | null>(null);
-  readonly type = input.required<CardType>();
+  readonly cardImgSrc = input.required<string>();
+  readonly nameTemplate = input<TemplateRef<any>>();
```

Comment:
why do you use an input, most of the time you should use a contentchild with template

Reply:
There are no performance issues. It's just that when you want to input a template, contentchild is more appropriate. But you can use an input without issue